addons/utm/models/utm_campaign.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import fields, models, api, SUPERUSER_ID
+
+
+class UtmCampaign(models.Model):
+    _name = 'utm.campaign'
+    _description = 'UTM Campaign'
+
+    name = fields.Char(string='Campaign Identifier', required=True, compute='_compute_name', store=True, readonly=False)
+    full_name = fields.Char(string='Campaign Name', required=True, translate=True)
+
+    user_id = fields.Many2one(
+        'res.users', string='Responsible',
+        required=True, default=lambda self: self.env.uid)
+    stage_id = fields.Many2one(
+        'utm.stage', string='Stage', ondelete='restrict', required=True,
+        default=lambda self: self.env['utm.stage'].search([], limit=1),
+        group_expand='_group_expand_stage_ids')
+    tag_ids = fields.Many2many(
+        'utm.tag', 'utm_tag_rel',
+        'tag_id', 'campaign_id', string='Tags')
+
+    is_auto_campaign = fields.Boolean(default=False, string="Automatically Generated Campaign", help="Allows us to filter relevant Campaigns")
+    color = fields.Integer(string='Color Index')
+
+    _sql_constraints = [
+        ('unique_name', 'UNIQUE(name)', 'The name must be unique'),
+    ]
+
+    @api.depends('full_name')
+    def _compute_name(self):
```

Comment:
Would instead generate the identifier based on the user-given name if identifier is not given. It is strange to do the inverse.